### PR TITLE
plugins/notebook-navigator: init

### LINKED ISSUE #2449: [PLUGIN REQUEST] NotebookNavigator.nvim
| Field    | Description |
|----------|-------------|
| Plugin   | NotebookNavigator.nvim  |
| Homepage | https://github.com/GCBallesteros/NotebookNavigator.nvim  |
| Nixpkgs  | `false` | 

## Extra Information

Notebook navigator allows VSCode-like cell execution from .py (or other) files without the need for a complete notebook environment. This functionality greatly increases my productivity and I'd love to see it as a part of nixvim!

### PR DESCRIPTION
**Pull request by GaetanLepage**
Add support for [NotebookNavigator.nvim](https://github.com/GCBallesteros/NotebookNavigator.nvim), a neovim plugin to navigate and execute code cells.

Fixes #2449

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -100,9 +100,9 @@ The tests are located in the [tests/test-sources](tests/test-sources) directory,
 
 Tests can either be a simple attribute set, or a function taking `{pkgs}` as an input. The keys of the set are configuration names, and the values are a nixvim configuration.
 
-You can specify the special `tests` attribute in the configuration that will not be interpreted by nixvim, but only the test runner. The following keys are available:
+You can specify the special `test` attribute in the configuration that will not be interpreted by nixvim, but only the test runner. The following keys are available:
 
-- `tests.dontRun`: avoid launching this test, simply build the configuration.
+- `test.runNvim`: Set to `false` to avoid launching nvim with this configuration and simply build the configuration.
```
Comment (khaneliman) [inline CONTRIBUTING.md:105]: Good catch, didn't notice we never updated this.

File: plugins/by-name/notebook-navigator/default.nix [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+          "molten"
+          "toggleterm"
+        ]
+        ''
+          The repl plugin with which to interface.
+          Current options:
+          - "iron" for iron.nvim,
+          - "toggleterm" for toggleterm.nvim,
+          - "molten" for molten-nvim
+          - or "auto" which checks which of the above are installed
```
Comment (khaneliman) [inline plugins/by-name/notebook-navigator/default.nix:69]:
```suggestion
          - "auto" which checks which of the above are installed
```

File: plugins/by-name/notebook-navigator/default.nix [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    repl_provider =
+      defaultNullOpts.mkEnumFirstDefault
+        [
+          "auto"
+          "iron"
+          "molten"
+          "toggleterm"
+        ]
+        ''
+          The repl plugin with which to interface.
```
Comment (khaneliman) [inline plugins/by-name/notebook-navigator/default.nix:64]:
```suggestion
          The repl plugin with which to interface.
          
```

File: plugins/by-name/notebook-navigator/default.nix [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+  settingsOptions = {
+    cell_markers = defaultNullOpts.mkAttrsOf' {
+      type = types.str;
+      pluginDefault = { };
+      example = {
+        python = "# %%";
+      };
+      description = ''
+        Code cell marker.
```
Comment (khaneliman) [inline plugins/by-name/notebook-navigator/default.nix:21]:
```suggestion
        Code cell marker.
        
```

File: plugins/by-name/notebook-navigator/default.nix [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        Cells start with the marker and end either at the beginning of the next cell or at the end of
+        the file.
```
Comment (khaneliman) [inline plugins/by-name/notebook-navigator/default.nix:23]:
```suggestion
        the file.
        
```

File: plugins/by-name/notebook-navigator/default.nix [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+          The repl plugin with which to interface.
+          Current options:
+          - "iron" for iron.nvim,
+          - "toggleterm" for toggleterm.nvim,
+          - "molten" for molten-nvim
+          - or "auto" which checks which of the above are installed
+        '';
+
+    syntax_highlight = defaultNullOpts.mkBool false ''
+      Syntax based highlighting.
```
Comment (khaneliman) [inline plugins/by-name/notebook-navigator/default.nix:73]:
```suggestion
      Syntax based highlighting.
      
```

**Review by khaneliman — COMMENTED** [review]
LGTM overall, I didn't pull down to double check the docs generation, but I think some of them need new lines to properly separate thoughts in the description.

**Review by khaneliman — APPROVED** [review]
LGTM thanks!

**Comment by GaetanLepage** [comment]
@mergify queue

**Pull request merged**